lib/portage/package/ebuild/doebuild.py

```diff
@@ -2133,9 +2185,38 @@ def spawn(
         logname_backup = mysettings.configdict["env"].get("LOGNAME")
         mysettings.configdict["env"]["LOGNAME"] = logname
 
+    eapi = mysettings["EAPI"]
+
+    unexported_env_vars = None
+    if "export-pms-vars" not in mysettings.features or not eapi_exports_pms_vars(eapi):
+        unexported_env_vars = _unexported_pms_vars
+
+    if unexported_env_vars:
+        orig_env = mysettings.environ()
+        # Copy since we are potentially removing keys from the dict.
+        env = orig_env.copy()
+
+        t = env["T"]
+        if not os.path.isdir(t):
+            os.makedirs(t)
+
+        ebuildExtraSource = os.path.join(t, ".portage-ebuild-extra-source")
```

Comment:
Note that the T directory is currently not ideal for the "depend" phase, since the parent build directory is not locked and cleaned up afterwards. Therefore, a file under T will not be cleaned up for the "depend" phase.

Reply:
Any recommendation of an alternative approach? Should and could we run the code creating the .portage-ebuild-extra-source file only of an ebuild phase function is going to be invoked afterwards?

Reply:
We could maybe

```python
        t = env["T"]
        if os.path.isdir(t):
            ebuild_extra_source_path = os.path.join(t, ".portage-ebuild-extra-source")
        else:
            ebuild_extra_source_path = tempfile.NamedTemporaryFile(
                suffix=f"portage-ebuild-extra-source",
                delete=True,
                delete_on_close=False,
            ).name
```